src/api/java/com/minecolonies/api/util/InventoryUtils.java

```diff
@@ -133,7 +133,7 @@ public static Item getItemFromBlock(final Block block)
     @NotNull
     public static Boolean isItemStackEmpty(@Nullable final ItemStack stack)
     {
-        return stack == null || stack == EMPTY || stack.getCount() <= 0;
+        return stack == null || stack == EMPTY || stack.getItem() != null || stack.getCount() <= 0;
```

Comment:
Nope ,`this.updateEmptyState();` gets called in all ItemStack constructors including the one from NBT.